backend/app/tests/services/users/test_utils.py

```diff
@@ -28,3 +37,150 @@ def test_check_creator_github_handle_not_in_list(self, github_data, projects, pu
         username = github_data(username="invalid")["username"]
         assert not any(username in project["creator_github_handle"] for project in projects())
         assert not any(username in publication["creator_github_handle"] for publication in publications())
+
+
+@pytest.fixture
+def mock_user():
+    return mUser(
+        id=1,
+        github_id=12345,
+        username="test_user",
+        email="test@example.com",
+        avatar_url=None,
+        role="reviewer",
+        created_on=datetime.datetime.utcnow(),
+        last_login=datetime.datetime.utcnow(),
+        is_active=True,
+        remarks=[],
+    )
+
+
+@pytest.mark.parametrize(
+    "id_, github_id, username, email, avatar_url, role, is_active",
+    [
+        (1, 123, "test_user", "test@example.com", None, "reviewer", True),
+        (2, 1234, "second_user", "second@example.com", "http://example.com/avatar2", "contributor", False),
+    ],
+)
+def test_get_user_with_mocked_db_session(mock_session, id_, github_id, username, email, avatar_url, role, is_active):
+    mock_session.query.return_value.filter.return_value.first.return_value = mUser(
+        id=id_,
+        github_id=github_id,
+        username=username,
+        email=email,
+        avatar_url=avatar_url,
+        role=role,
+        is_active=is_active,
+        created_on=datetime.datetime.utcnow(),
+        last_login=datetime.datetime.utcnow(),
+        remarks=[],
+    )
+
+    result_user = get_user(github_id=github_id)
+
+    assert isinstance(result_user, User)
+    assert result_user.github_id == github_id
+    assert result_user.id == id_
+    assert result_user.username == username
+    assert result_user.email == email
+    assert result_user.avatar_url == avatar_url
+    assert result_user.role == role
+    assert result_user.is_active == is_active
+
+
+def test_is_user_active(mock_user):
+    assert is_user_active(mock_user)
+
+
+def test_get_roles():
+    assert get_roles() == ["administrator", "superuser", "writer", "reviewer"]
+
+
+def test_update_user_role_update(mock_session, mock_user):
+    mock_session.query.return_value.filter.return_value.first.return_value = mock_user
+    mock_session.commit.return_value = None
+
+    user_data = {
+        "id": mock_user.id,
+        "github_id": mock_user.github_id,
+        "username": mock_user.username,
+        "email": mock_user.email,
+        "avatar_url": mock_user.avatar_url,
+        "role": "administrator",
+        "is_active": mock_user.is_active,
+        "created_on": mock_user.created_on,
+        "last_login": mock_user.last_login,
+        "remarks": [],
+    }
```

Comment:
Whenever you are repeating this, perhaps you could try something like: 
`user_data = {key: value for key, value in vars(mock_user).items() if not key.startswith('_')}`
if there is a need you could just change the `role`:
`user_data["role"]= Role.ADMIN.value`